add new 'size-limit' option for type 'boot-emmc'

ejoerns opened the pull request:
For special cases, where e.g. the manufacturer decided to store data at the end of the boot partition and where this information is required and cannot be migrated, we allow specifying a `size-limit` option for the `boot-emmc` slot type that allows limiting the cleared/written size.

Fixes: #780

jluebbe reviewed (changes requested):
Looks good otherwise, but test coverage would be nice.

ejoerns commented on src/update_handler.c:
```diff
@@ -32,21 +32,52 @@ GQuark r_update_error_quark(void)
+ * image is checked against this limit first.
+ *
+ * @param fd file descriptor of device to check
+ * @param slot slot for optional size-limit check, or NULL
```
Thought about this, too. But decided on this solution with the least impact. Can change it, however.

ejoerns commented:
v2:

* rebased onto master
* added suggestions
* `slot` parameter of `check_image_size()` is now mandatory and always set
* `g_message()` log level for `clear_slot()` information if `size-limit` is set. This improves visibility and helps testing.
* two `size-limit` test cases in `test_write_slot.py`
